src/linux/InotifyTree.cpp

```diff
@@ -334,34 +344,8 @@ bool InotifyTree::InotifyNode::inotifyInit() {
     return false;
   }
 
-  auto *childrenToRemove = new std::vector<std::string>;
-  childrenToRemove->reserve(mChildren->size());
-  for (auto i = mChildren->begin(); i != mChildren->end(); ++i) {
-    if (!i->second->inotifyInit()) {
-      childrenToRemove->push_back(i->second->getName());
-    }
-  }
-
-  if (childrenToRemove->size() > 0) {
-    struct stat file;
-
-    if (
-      stat(mFullPath.c_str(), &file) < 0 ||
-      !S_ISDIR(file.st_mode)
-    ) {
-      mAlive = false;
-    } else {
-      for (auto i = childrenToRemove->begin(); i != childrenToRemove->end(); ++i) {
-        removeChild(*i);
-      }
-      mWatchDescriptorInitialized = true;
-      mTree->addNodeReferenceByWD(mWatchDescriptor, this);
-    }
-  } else {
-    mWatchDescriptorInitialized = true;
-    mTree->addNodeReferenceByWD(mWatchDescriptor, this);
-  }
-  delete childrenToRemove;
```

Comment:
This is a lot of cleanup code to remove. Can you explain your reasoning for removing this block, please?

Reply:
Sure, this code was part of the logic required to do the recursive `inotifyInit` after having built the tree using `scandir` before. My change now does `inotifyInit` as one of the first steps of the recursive tree building. See https://github.com/Axosoft/nsfw/blob/f029e51e166d7ff94f471bc2f061dc2621c5f312/src/linux/InotifyTree.cpp#L157-L186

Reply:
Please tell me if I misunderstood something with the previous logic that I forgot to keep included in this change.

Reply:
I think I was originally trying to spec out that we never create an inotify handle for anything that isn't a directory. By inverting this structure, we would need to be certain that inotify_add_watch does not create a handle for non-files, otherwise we'll burn through our inotify limit much faster than we used to.

Reply:
Likely need to rig up https://man7.org/linux/man-pages/man2/inotify_add_watch.2.html `IN_ONLYDIR` to the mask to enforce that behavior after this change. Here's some more reading on that https://man7.org/linux/man-pages/man7/inotify.7.html

Reply:
We can add `IN_ONLYDIR` indeed, but from what I understood `InotifyNode` instances are only created on directories: https://github.com/Axosoft/nsfw/blob/f029e51e166d7ff94f471bc2f061dc2621c5f312/src/linux/InotifyTree.cpp#L217-L233

Reply:
Gotcha, and we're able to determine if it is a directory initially from the InotifyEventLoop. Thanks.